pyvista/core/datasetattributes.py

```diff
@@ -288,7 +291,12 @@ def pop(self, key):
             copy = vtk_arr.NewInstance()
             copy.DeepCopy(vtk_arr)
             vtk_arr = copy
-        self.remove(key)
+        try:
+            self.remove(key)
+        except KeyError:
+            if default in self.pop.__defaults__:
+                raise KeyError
```

Comment:
```suggestion
                raise
```
Lazily commiting here as I have other branches up...